lib/main.js

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2012-2016 Digital Bazaar, Inc. All rights reserved.
+ */
+/* jshint node: true */
+'use strict';
+
+var _ = require('lodash');
+var bedrock = require('bedrock');
+var BedrockError = bedrock.util.BedrockError;
+var config = bedrock.config;
+var brPermission = require('bedrock-permission');
+var brPassport = require('bedrock-passport');
+var database = require('bedrock-mongodb');
+var ensureAuthenticated = brPassport.ensureAuthenticated;
+var validate = require('bedrock-validation').validate;
+
+require('bedrock-express');
+
+// load config
+require('./config');
+
+bedrock.events.on('bedrock.configure', function() {
+  config.permission.roleBaseUrl =
+    config.server.baseUri + config['permission-http'].routes.roles;
+});
+
+bedrock.events.on('bedrock-express.configure.routes', function(app) {
+  var permissionsBasePath = config['permission-http'].routes.permissions;
+  var permissionsBaseUrl = config.server.baseUri + permissionsBasePath;
+  var rolesBasePath = config['permission-http'].routes.roles;
+
+  /**
+   * Retrieves all permissions from the config held permissions.
+   */
+  app.get(permissionsBasePath, ensureAuthenticated, function(req, res, next) {
+    var permissions = config.permission.permissions;
+    res.json(permissions);
```

Comment:
cc: @davidlehn -- Do you think we should put any authentication restrictions on getting all permissions in the system? People can already see their own permissions (but not more) via the session service. I think we should probably lock this down to someone with the PERMISSION_ADMIN permission. Being able to see all permissions on the system gives you hints about what's installed.

Thoughts?

Does anyone know if this would cause a problem?